src/client/QXmppCall_p.h

```diff
@@ -0,0 +1,133 @@
+        int pt;
+        QString name;
+        int channels;
+        int clockrate;
+        QString gstPay;
+        QString gstDepay;
+        QString gstEnc;
+        QString gstDec;
+        struct Property {
+            QString name;
+            int value;
+        };
+        // Use e.g. gst-inspect-1.0 x264enc to find good encoder settings for live streaming
+        QList<Property> encProps;
+    };
+
+    QXmppCallPrivate(QXmppCall *qq);
+    ~QXmppCallPrivate();
+
+    void ssrcActive(uint sessionId, uint ssrc);
+    void padAdded(GstPad *pad);
+    GstCaps * ptMap(uint sessionId, uint pt);
+    bool isFormatSupported(const QString &codecName) const;
+    void filterGStreamerFormats(QList<GstCodec> &formats);
+
+    QXmppCallStream *createStream(const QString &media, const QString &creator, const QString &name);
+    QXmppCallStream *findStreamByMedia(const QString &media);
+    QXmppCallStream *findStreamByName(const QString &name);
+    QXmppCallStream *findStreamById(const int id);
+    QXmppJingleIq::Content localContent(QXmppCallStream *stream) const;
+
+    void handleAck(const QXmppIq &iq);
+    bool handleDescription(QXmppCallStream *stream, const QXmppJingleIq::Content &content);
+    void handleRequest(const QXmppJingleIq &iq);
+    bool handleTransport(QXmppCallStream *stream, const QXmppJingleIq::Content &content);
+    void setState(QXmppCall::State state);
+    bool sendAck(const QXmppJingleIq &iq);
+    bool sendInvite();
+    bool sendRequest(const QXmppJingleIq &iq);
+    void terminate(QXmppJingleIq::Reason::Type reasonType);
+
+    QXmppCall::Direction direction;
+    QString jid;
+    QString ownJid;
+    QXmppCallManager *manager;
+    QList<QXmppJingleIq> requests;
+    QString sid;
+    QXmppCall::State state;
+
+    GstElement *pipeline;
+    GstElement *rtpbin;
+
+    // Media streams
+    QList<QXmppCallStream*> streams;
+    int nextId;
+
+    // Supported codecs
+    QList<GstCodec> videoCodecs = {
+        {.pt = 100, .name = "H264", .channels = 1, .clockrate = 90000, .gstPay = "rtph264pay", .gstDepay = "rtph264depay", .gstEnc = "x264enc", .gstDec = "avdec_h264", .encProps = {{"tune", 4}, {"speed-preset", 1}, {"bitrate", 256}}},
+        {.pt = 99, .name = "VP8", .channels = 1, .clockrate = 90000, .gstPay = "rtpvp8pay", .gstDepay = "rtpvp8depay", .gstEnc = "vp8enc", .gstDec = "vp8dec", .encProps = {{"deadline", 20000}, {"target-bitrate", 256000}}},
+        // vp9enc and x265enc seem to be very slow. Give them a lower priority for now.
+        {.pt = 102, .name = "H265", .channels = 1, .clockrate = 90000, .gstPay = "rtph265pay", .gstDepay = "rtph265depay", .gstEnc = "x265enc", .gstDec = "avdec_h265", .encProps = {{"tune", 4}, {"speed-preset", 1}, {"bitrate", 256}}},
+        {.pt = 101, .name = "VP9", .channels = 1, .clockrate = 90000, .gstPay = "rtpvp9pay", .gstDepay = "rtpvp9depay", .gstEnc = "vp9enc", .gstDec = "vp9dec", .encProps = {{"deadline", 20000}, {"target-bitrate", 256000}}}
```

Comment:
I'm not sure if I understand the priorities here, but shouldn't the value (`.pt`) here be lower than the values of H.264/VP8 to not prefer them?

Otherwise, this would mean that PCM, AAC and Speex are preferred over Opus which is probably not a good idea.

Reply:
The pt value does not matter for the priorities. Only the order in our list / the order of payload-type entries in the containing xmpp message determines the priority.

When it comes to pt values we only have to make sure to follow the standardized values if available and otherwise have a unique 1-1 mapping.

Reply:
Ah, sorry then :)